docs/dev/guidelines/client-tests.rst

```diff
@@ -0,0 +1,257 @@
+******
+Client Tests
+******
+
+**If you are new to client testing, it is highly recommended that you work through the testing part of the angular tutorial:** https://angular.io/guide/testing
+
+We use Jest (https://jestjs.io/) as our client testing framework.
+
+There are different tools available to support client testing. A common combination you can see in our codebase is:
+
+- Sinon (https://sinonjs.org/) for creating test spies, stubs and mocks
+- Chai (https://www.chaijs.com/) with Sinon Chai (https://github.com/domenic/sinon-chai) for assertions.
+- NgMocks (https://www.npmjs.com/package/ng-mocks) for mocking the dependencies of an angular component.
```

Comment:
I think the instructions are more readable, if you use hyperlinks on the keywords instead of putting the whole link as text. Or is there a specific reason?

Reply:
```suggestion
**If you are new to client testing, it is highly recommended that you work through the** `testing part <https://angular.io/guide/testing>`_ **of the angular tutorial:**

We use `Jest <https://jestjs.io>`_ as our client testing framework.

There are different tools available to support client testing. A common combination you can see in our codebase is:

- `Sinon <https://sinonjs.org/>`_ for creating test spies, stubs and mocks
- `Chai <https://www.chaijs.com/>`_ with `Sinon Chai <https://github.com/domenic/sinon-chai>`_ for assertions.
- `NgMocks <https://www.npmjs.com/package/ng-mocks/>`_ for mocking the dependencies of an angular component.

```

Reply:
if you accept these changes, you should probably change the pattern throughout the course of this file and and change this at other places as well :) e.g. for the sandbox link

Reply:
I don't think there is a specific reason to do it this way. I will apply your changes together with this pattern throughout the course.